uast/uast-java/src/org/jetbrains/uast/java/expressions/JavaUClassLiteralExpression.kt

```diff
@@ -17,15 +17,18 @@ package org.jetbrains.uast.java
 
 import com.intellij.psi.PsiClassObjectAccessExpression
 import com.intellij.psi.PsiType
+import org.jetbrains.annotations.ApiStatus
 import org.jetbrains.uast.UClassLiteralExpression
 import org.jetbrains.uast.UElement
+import org.jetbrains.uast.UTypeReferenceExpression
 
+@ApiStatus.Internal
 class JavaUClassLiteralExpression(
   override val sourcePsi: PsiClassObjectAccessExpression,
   givenParent: UElement?
 ) : JavaAbstractUExpression(givenParent), UClassLiteralExpression {
   override val type: PsiType
     get() = sourcePsi.operand.type
 
-  override val expression: JavaUTypeReferenceExpression by lz { JavaUTypeReferenceExpression(sourcePsi.operand, this) }
```

Comment:
also there is a known external usage ((

Reply:
That's bad... :( there is nothing more a user can get learn from the return type here. Left TODO as well.